igvm/vm.py

```diff
@@ -151,6 +151,23 @@ def set_memory(self, memory):
         """Resizes the host memory."""
         self.hypervisor.vm_set_memory(self, memory)
 
+    def set_hostname(self, new_hostname, transaction=None):
+        """Changes the hostname of a VM"""
+        self.previous_hostname = self.dataset_obj['hostname']
```

Comment:
i really don't like this. please try to pass this information in the `on_rollback` callback already. you can do something like `transaction.on_rollback('reset_hostname', lambda: self.revert_hostname(self.dataset_obj['hostname']))`. obviousely before you change it in the `dataset_obj`. or save it to `previous_hostname` at least. but there is no reason to make this an instance variable is there? just my opinion

Reply:
or rather don't use lambdas and use the `args`/`kwargs` of `on_rollback`

Reply:
I did this way to try to be consistent with what `reset_state` was already doing. 

I can move it back into the `rename` function also

Reply:
ah i see. then i am fine with it for now. this is not a blocker